### PR TITLE
refactor(publish): derive outer publish timeout from per-relay fan-out (#3688)

### PR DESCRIPTION
**Pull request by realmeylisdev**
## Description

Replaces the implicit `5s × 6 relay = 30s` coupling left in #3683 with an
encoded contract:

- **`nostr_sdk`** exports `RelayPool.perRelaySendTimeout` (new public
  `static const Duration`) so callers wrapping publish calls in their
  own outer guard can size that guard against the worst-case sequential
  fan-out instead of duplicating the literal.
- **`VideoEventPublisher`** now derives the outer `Future.timeout` via a
  top-level `outerPublishTimeoutFor(int relayCount)` helper —
  `perRelay × configuredRelayCount + buffer`, clamped to a `[10s, 60s]`
  range. Hard-coding the outer 30s would silently break the moment a
  user adds a relay or the SDK retunes the per-relay cap; the
  derivation encodes the invariant in code.
- **Log enrichment** (sub-item 3 of #3688): per-relay timeouts inside
  `_sendCollect` now log `relay.url`, `connected` state, and `authed`
  state. The existing outer timeout log gains `configuredRelayCount` so
  a coupling-violation incident is visible from logs alone.

**Decision on telemetry** (sub-item 3 of #3688): logging only, no new
analytics SDK. The repo's existing `Log.error` flows to the production
sink already in place; a histogram pipeline would require an analytics
seam this codebase doesn't currently have. Reconsider if the per-relay
timeout starts firing in the field.

**Related Issue:** Relates to #3688 (closes sub-items 1 and 3; sub-item
2 — Keycast `sign_canonical` cross-repo tracking — opened at
[divinevideo/keycast#186](https://github.com/divinevideo/keycast/issues/186)
with the determinism contract and a concrete cross-implementation test
vector).

**Stacked on:** #3683 — base branch is `fix/publish-hang-disconnected-relay`.

## Type of Change

- [x] 🧹 Code refactor

## Trade-off worth surfacing

The 60s ceiling means the strict `outer > inner_worst_case + buffer`
invariant evaporates at `relayCount == 12` (zero buffer) and inverts at
`relayCount >= 13` (outer < inner). This re-introduces the original
false-negative-publish failure mode for the unlikely edge case of 13+
relays all simultaneously wedged-after-handshake. Documented loudly on
`_outerPublishTimeoutCeiling` and pinned by a dedicated test
(`'invariant degrades at the ceiling boundary (relayCount >= 12)'`).
The retry loop in `publishDirectUpload` absorbs the rare
false-negative when it does happen. Alternatives considered (no
ceiling = 4-minute spinner on misconfigured 50-relay sets; parallel
fan-out = separate ticket already noted in #3683 out-of-scope).

## Test coverage

| File | What it locks |
|---|---|
| `mobile/test/services/video_event_publisher_test.dart` | `outerPublishTimeoutFor` math: floor / linear scaling / ceiling clamp / strict-exceed-up-to-boundary / boundary degradation at relayCount=12 and 13. |
| `mobile/test/services/video_event_publisher_publish_confirmation_test.dart` | `currentOuterPublishTimeout` getter wiring: reflects helper, updates live with relay count, never collapses to a single literal (regression guard against reverting to a hard-coded `Duration(seconds: 30)`). |
| `mobile/packages/nostr_sdk/test/unit/relay_pool_concurrency_test.dart` | `RelayPool.perRelaySendTimeout` exposed as a stable public contract; existing per-relay timeout test now references the constant instead of the 5s literal. |

89 tests pass on push; `flutter analyze` clean on `mobile/lib test integration_test` and `mobile/packages/nostr_sdk`; `dart format` clean.

## Out of scope (tracked elsewhere)

- Backend implementation of `sign_canonical` — tracked at [divinevideo/keycast#186](https://github.com/divinevideo/keycast/issues/186) (#3688 sub-item 2; mobile is already wired and forward-compatible).
- Parallel fan-out inside `_sendCollect` — eliminates the coupling
  entirely. Listed as a future follow-up in #3683's out-of-scope
  section; explicitly not bundled here.

## Test plan

- [ ] CI: `Tests`, `Analyze`, `Format`, `Generated Files`, `build / build` (divine_ui coverage) green.
- [ ] Stacked PR base shows correctly as `fix/publish-hang-disconnected-relay`.
- [ ] `git diff main..HEAD -- mobile/lib/services/video_event_publisher.dart` shows the helper extraction and getter, not just isolated literal swaps.
- [ ] Manual smoke (after #3683 merges and this rebases on main): publish a clip with all 6 relays healthy → completes within ~35s. Add 2 unreachable relays (8 total) → publish still completes under ~45s thanks to `skipReconnect`; outer guard does not fire.

### CODE REVIEW

File: mobile/packages/nostr_sdk/lib/relay/relay_pool.dart [hunk 1]
```diff
@@ -31,6 +31,16 @@ class RelayPool {
...
 
+  /// Per-relay cap inside [_sendCollect]'s sequential fan-out.
+  ///
+  /// Exposed as part of the public API so callers that wrap publish calls
+  /// in their own outer guard can size that guard against the worst-case
+  /// fan-out duration (`perRelaySendTimeout * configuredRelays.length`)
+  /// rather than duplicating the literal. See
+  /// `mobile/lib/services/video_event_publisher.dart` for the canonical
+  /// caller-side derivation.
+  static const Duration perRelaySendTimeout = Duration(seconds: 5);
```
Comment (NotThatKindOfDrLiz) [inline mobile/packages/nostr_sdk/lib/relay/relay_pool.dart:42]: Good seam to expose publicly: this gives outer publish guards a single source of truth for the sequential fan-out cap instead of making app-layer callers mirror SDK internals.

File: mobile/lib/services/video_event_publisher.dart [hunk 2]
```diff
@@ -27,6 +28,77 @@ import 'package:openvine/utils/proofmode_publishing_helpers.dart';
...
+
+/// Buffer added on top of the per-relay × count derivation. Covers the
+/// microtask queue drains between sequential `relay.send` calls inside
+/// [RelayPool._sendCollect], plus a small allowance for log formatting
+/// and other in-process scheduling jitter. Picked at one
+/// `perRelaySendTimeout` worth of slack — small relative to the total
+/// `perRelay × N` budget at default-config sizes (≈14% of the 35s outer
+/// at N=6) but large enough to absorb realistic dispatch overhead on
+/// cold-start without erosion as the relay count grows.
+const Duration _outerPublishTimeoutBuffer = RelayPool.perRelaySendTimeout;
```
Comment (NotThatKindOfDrLiz) [inline mobile/lib/services/video_event_publisher.dart:67]: This follow-up finishes the coupling cleanup: the caller-side slack is now derived from the SDK-side per-relay send cap instead of duplicating a `5s` literal here.

**Review by NotThatKindOfDrLiz — APPROVED** [review]
The rebased PR is in good shape. The outer publish timeout is now derived from the SDK-side per-relay send cap instead of being implicitly coupled to a hard-coded 30s, the ceiling trade-off is documented and pinned by tests, and the timeout logs now carry the context you would want if this ever fires in production.

The last small cleanup is in too: the caller-side buffer now references RelayPool.perRelaySendTimeout directly, so the remaining 5s coupling is gone as well.
